table/tables/index.go

```diff
@@ -103,236 +103,299 @@ func (c *index) GenIndexValue(sc *stmtctx.StatementContext, distinct bool, index
 	return tablecodec.GenIndexValuePortal(sc, c.tblInfo, c.idxInfo, c.needRestoredData, distinct, false, indexedValues, h, c.phyTblID, restoredData)
 }
 
+func (c *index) getIndexedValue(indexedValues []types.Datum) [][]types.Datum {
+	if !c.idxInfo.MVIndex {
+		return [][]types.Datum{indexedValues}
+	}
+
+	vals := make([][]types.Datum, 0, 16)
+	jsonIdx := 0
+	jsonIsNull := false
+	existsVals := make(map[string]struct{})
+	var buf []byte
+	for !jsonIsNull {
+		val := make([]types.Datum, 0, len(indexedValues))
+		for i, v := range indexedValues {
+			if !c.tblInfo.Columns[c.idxInfo.Columns[i].Offset].FieldType.IsArray() {
+				val = append(val, v)
+			} else {
+				if v.IsNull() {
+					val = append(val, v)
+					jsonIsNull = true
+					jsonIdx++
+					continue
+				}
+				elemCount := v.GetMysqlJSON().GetElemCount()
```

Comment:
Can a MVIndex have multiple JSON-ARRAY columns like `index idx((cast(a->'$.COL1' as signed array)), (cast(a->'$.COL2' as signed array)))`

Reply:
no, only 1 JSON-ARRAY is allowed